BenchmarkTests/Benchmarks/Sources/MetricExporter.swift

```diff
@@ -0,0 +1,159 @@
+        let interval: Int64?
+        let metric: String
+        let unit: String?
+        let points: [Point]
+        let resources: [Resource]
+        let tags: [String]
+    }
+
+    let encoder = JSONEncoder()
+    let configuration: Configuration
+
+    // swiftlint:disable force_unwrapping
+    let intake = URL(string: "https://api.datadoghq.com/api/v2/series")!
+    let prefix = "{ \"series\": [".data(using: .utf8)!
+    let separator = ",".data(using: .utf8)!
+    let suffix = "]}".data(using: .utf8)!
+    // swiftlint:enable force_unwrapping
+
+    required init(configuration: Configuration) {
+        self.configuration = configuration
+        super.init()
+    }
+
+    func export(metrics: [Metric], shouldCancel: (() -> Bool)?) -> MetricExporterResultCode {
+        do {
+            let series = try metrics.map(transform)
+            try submit(series: series)
+            return.success
+        } catch {
+            return .failureNotRetryable
+        }
+    }
+    
+    /// Transforms otel `Metric` to Datadog `serie`.
+    ///
+    /// - Parameter metric: The otel metric
+    /// - Returns: The timeserie.
+    func transform(_ metric: Metric) throws -> Serie {
+        var tags: Set<String> = []
+
+        let points: [Serie.Point] = try metric.data.map { data in
+            let timestamp = Int64(data.timestamp.timeIntervalSince1970)
+
+            data.labels.forEach { tags.insert("\($0):\($1)") }
+
+            switch data {
+            case let data as SumData<Double>:
+                return Serie.Point(timestamp: timestamp, value: data.sum)
+            case let data as SumData<Int>:
+                return Serie.Point(timestamp: timestamp, value: Double(data.sum))
+            case let data as SummaryData<Double>:
+                return Serie.Point(timestamp: timestamp, value: data.sum)
+            case let data as SummaryData<Int>:
+                return Serie.Point(timestamp: timestamp, value: Double(data.sum))
+//            case let data as HistogramData<Int>:
+//                return Serie.Point(timestamp: timestamp, value: Double(data.sum))
+//            case let data as HistogramData<Double>:
+//                return Serie.Point(timestamp: timestamp, value: data.sum)
+            default:
+                throw MetricExporterError.unsupportedMetric(
+                    aggregation: metric.aggregationType,
+                    dataType: type(of: data)
+                )
+            }
+        }
+
+        return Serie(
+            type: MetricType(metric.aggregationType),
+            interval: nil,
+            metric: metric.name,
+            unit: nil,
+            points: points,
+            resources: [],
+            tags: Array(tags)
+        )
+    }
+    
+    /// Submit timeseries to the Metrics intake.
+    ///
+    /// - Parameter series: The timeseries.
+    func submit(series: [Serie]) throws {
+        var data = try series.reduce(Data()) { data, serie in
```

Comment:
Should we expect large payloads here? If so, what happens if it exceeds a certain size?

Reply:
There is no caching so we will upload 3 metrics per payload, with only one point per metrics. Each upload will happen every 10 seconds as configured [here](https://github.com/DataDog/dd-sdk-ios/pull/1993/files#diff-b5ae3e89d1c4c79036d3ea45588a247fff697d5ccf90f172c0be2856a4542e79R92). So each payload will be very small 👍